src/todos/todos.service.ts

```diff
@@ -1,4 +1,40 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable } from "@nestjs/common";
+import { InjectModel } from "@nestjs/sequelize";
+import { CreateToDoDto } from "./dto/create-todo.dto";
+import { Todo } from "./todos.model";
 
 @Injectable()
-export class TodosService {}
+export class TodosService {
+  constructor(@InjectModel(Todo) private todoRepository: typeof Todo) {}
+
+  async createToDo(dto: CreateToDoDto) {
+    const todo = await this.todoRepository.create(dto);
+    return todo;
+  }
+
+  async getAllTodos() {
+    const todos = await this.todoRepository.findAll();
+    return todos;
+  }
+
+  async getTodo(id: number) {
+      
+    if (!id) return null;
+    const todo = await this.todoRepository.findOne({where: {id}});
+    return todo;
+  }
+
+  async updateTodo(id: number, dto: CreateToDoDto) {
```

Comment:
It is not CreateToDoDto if you use it as an update. Inherit from Create and add id there.

Reply:
In this case service method shares the same dto (data transfer object) with creteToDo (because they are same), but i agree it should be better practise to create another dto for this purpose.